lib/response/get/checkCacheAndReport.js

```diff
@@ -63,13 +62,10 @@ module.exports = function checkCacheAndReport(model, requestedPaths, observer,
         }
     }
 
-    // If there are values to report, then report.
-    // Which are under two conditions:
-    // 1.  This request is progressive
-    //
-    // 2.  The request if finished and the json key off
-    // the valueNode has a value.
-    if (progressive || ((progressive && results.hasValues || !progressive) && completed && valueNode !== undefined)) {
+    // Report locally available values if:
+    // - the request is in progressive mode, or
+    // - the request is complete and values were found
+    if (progressive || (completed && valueNode !== undefined)) {
```

Comment:
Unrelated to the PR but I've been eyeing this condition for cleanup. They're equivalent.